core/src/main/java/com/sap/ai/sdk/core/commons/ClientStreamingHandler.java

```diff
@@ -0,0 +1,93 @@
+package com.sap.ai.sdk.core.commons;
+
+import static com.sap.ai.sdk.core.JacksonConfiguration.getDefaultObjectMapper;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.function.BiFunction;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+
+/**
+ * Parse incoming JSON responses and handles any errors. For internal use only.
+ *
+ * @param <D> The type of the response.
+ * @param <E> The type of the exception to throw.
+ * @since 1.1.0
+ */
+@Slf4j
+public class ClientStreamingHandler<D extends StreamedDelta, E extends ClientException> {
+
+  @Nonnull private final Class<D> deltaType;
+  @Nonnull private final ClientResponseHandler<? extends ClientError, E> errorHandler;
+  @Nonnull private final BiFunction<String, Throwable, E> exceptionType;
+
+  /** The parses for JSON responses, will be private once we can remove mixins */
+  @Nonnull private ObjectMapper JACKSON = getDefaultObjectMapper();
```

Comment:
(Minor/Comment)

Considering the fields are equivalent in both classes:
* ClientResponseHandler
* ClientStreamingHandler

(, and the latter accesses methods from the former)

... maybe it would be worthwhile to check whether scenario `ClientStreamingHandler extends ClientResponseHandler` potentially simplifies the code even further, while reducing code duplication.

Reply:
Done